synapse/config/server.py

```diff
@@ -368,14 +368,14 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
 
         # Whether to enable user presence.
         presence_config = config.get("presence") or {}
-        self.use_presence = presence_config.get("enabled")
-        if self.use_presence is None:
-            self.use_presence = config.get("use_presence", True)
-
-        # Selectively enable syncing of presence, even if it is disabled.
-        self.use_presence_for_sync = presence_config.get(
-            "enabled_for_sync", self.use_presence
-        )
+        presence_enabled = presence_config.get("enabled")
+        if presence_enabled is None:
+            presence_enabled = config.get("use_presence", True)
```

Comment:
Can we complain if users give us an inappropriate value?

E.g. 

```python
if presence_enabled not in (True, False, None, "untracked"):
   raise ConfigError(...)
```

I'd also be happy to punt this to  #12651 if you'd rather keep this PR small.

Reply:
I'm afraid that people have old configs that this will now start erroring on, e.g. using 1 or something.

(Right now unless someone has put the special value "untracked", which I find unlikely, then things will still work OK.)

Reply:
Sad, but fair!